src/core/agent/services/signifyNotificationService.test.ts

```diff
@@ -0,0 +1,89 @@
+import { Agent } from "@aries-framework/core";
+import { SignifyNotificationService } from "./signifyNotificationService";
+
+const agent = jest.mocked({
+  modules: {
+    generalStorage: {
+      getAllAvailableIdentifierMetadata: jest.fn(),
+      getAllArchivedIdentifierMetadata: jest.fn(),
+      getIdentifierMetadata: jest.fn(),
+      saveIdentifierMetadataRecord: jest.fn(),
+      archiveIdentifierMetadata: jest.fn(),
+      deleteIdentifierMetadata: jest.fn(),
+      updateIdentifierMetadata: jest.fn(),
+      getKeriIdentifiersMetadata: jest.fn(),
+    },
+    signify: {
+      getIdentifierByName: jest.fn(),
+      createIdentifier: jest.fn(),
+      getAllIdentifiers: jest.fn(),
+      markNotification: jest.fn(),
+    },
+  },
+  dids: {
+    getCreatedDids: jest.fn(),
+    resolve: jest.fn(),
+    create: jest.fn(),
+  },
+  genericRecords: {
+    save: jest.fn(),
+    findAllByQuery: jest.fn(),
+    findById: jest.fn(),
+    deleteById: jest.fn(),
+  },
+});
+
+const signifyNotificationService = new SignifyNotificationService(
+  agent as any as Agent
+);
+
+describe("Signify notification service of agent", () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  test("callback should be called when there are KERI notifications", async () => {
+    const callback = jest.fn();
+    const notes = [
+      {
+        i: "string",
+        dt: "string",
+        r: false,
+        a: {
+          r: "/multisig/icp",
+          d: "string",
+          m: "",
+        },
+      },
+      {
+        i: "string",
+        dt: "string",
+        r: false,
+        a: {
+          r: "unknown",
+          d: "string",
+          m: "",
+        },
+      },
+      {
+        i: "string",
+        dt: "string",
+        r: false,
+        a: {
+          r: "/exn/ipex/grant",
+          d: "string",
+          m: "",
+        },
+      },
+    ];
+    agent.genericRecords.save = jest
+      .fn()
+      .mockReturnValue({ id: "id", createdAt: new Date(), content: {} });
+    jest.useFakeTimers();
+    for (const notif of notes) {
+      await signifyNotificationService.processNotification(notif, callback);
+    }
+    expect(agent.genericRecords.save).toBeCalledTimes(2);
+    expect(callback).toBeCalledTimes(2);
```

Comment:
Test should have a third notification type that we don't process, to ensure `callback` and `genericRecords.save` are still only called twice

Reply:
I already have that in the index 1 of the notes array.

Reply:
Whoops, read that too quickly, my mistake apologies! thanks :)